### PR TITLE
Junction tree bernstein copula

### PR DESCRIPTION
**Pull request by regislebrun**
This PR provides the JunctionTreeBernsteinCopula class. For now, it implements only computePDF() and getRealization(), but it is already useful for Monte Carlo sampling and minimum volume set computation (eg for anomaly detection).
It also fixes some const correctness issues in the JunctionTree class.

### CODE REVIEW

File: python/src/JunctionTreeBernsteinCopula_doc.i.in [hunk 1]
```diff
@@ -0,0 +1,67 @@
+%feature("docstring") OTAGRUM::JunctionTreeBernsteinCopula
+"JunctionTreeBernsteinCopula distribution
+
```
Comment (jschueller) [inline python/src/JunctionTreeBernsteinCopula_doc.i.in]: Your commit comment could be useful here:
"This class allows to build a nonparametric copula from a given sample, a bin count and a junction tree descibing the conditional independencies in the data. It is based on a Bernstein estimation of each sub-copulas of the model."

File: lib/src/otagrum/JunctionTreeBernsteinCopula.hxx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
```
Comment (jschueller) [inline lib/src/otagrum/JunctionTreeBernsteinCopula.hxx]: Could you use the same headers as in master ? The date, authors and license is a bit different (2018/ LIP6/ GPL)

File: python/src/JunctionTreeBernsteinCopula_doc.i.in (hunk 1 above)
Comment (regislebrun) [reply python/src/JunctionTreeBernsteinCopula_doc.i.in]: done

File: lib/src/otagrum/JunctionTreeBernsteinCopula.hxx (hunk 2 above)
Comment (regislebrun) [reply lib/src/otagrum/JunctionTreeBernsteinCopula.hxx]: done

**Comment by regislebrun** [comment]
@phwuil Time to experiment different strategies to order the loop over cliques and separators to speed-up getRealization() ;-)!

**Pull request merged**